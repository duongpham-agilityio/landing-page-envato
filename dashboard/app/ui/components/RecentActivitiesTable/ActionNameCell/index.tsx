// Libs
import { memo } from 'react';
import { Flex, Td, Text, Tooltip } from '@chakra-ui/react';
import { formatUppercaseFirstLetter } from '@/lib/utils';

interface ActionNameCellProps {
  actionName: string;
}

const ActionNameCell = ({ actionName }: ActionNameCellProps): JSX.Element => (
  <Td
    py={5}
    pr={5}
    pl={0}
    fontSize="md"
    color="text.primary"
    fontWeight="semibold"
    textAlign="left"
    w={{ base: 150, md: 250, '6xl': 300 }}
  >
    <Flex alignItems="center" gap="10px">
      <Tooltip minW="max-content" placement="bottom-start" label={actionName}>
        <Text
          display="block"
          fontSize="md"
          fontWeight="semibold"
          wordBreak="break-all"
          textOverflow="ellipsis"
          overflow="hidden"
          pr={10}
          flex={1}
          w={{ base: 150, md: 250, '6xl': 300 }}
        >
          {formatUppercaseFirstLetter(`${actionName}`)}
        </Text>
      </Tooltip>
    </Flex>
  </Td>
);

const ActionNameCellMemorized = memo(ActionNameCell);

export default ActionNameCellMemorized;
