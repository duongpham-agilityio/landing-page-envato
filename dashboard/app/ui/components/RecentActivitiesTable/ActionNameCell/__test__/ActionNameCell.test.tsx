import { render } from '@testing-library/react';
import { Table } from '@chakra-ui/react';

// Components
import ActionNameCell from '..';

const mockProps = {
  actionName: 'Login',
};

describe('ActionNameCell component', () => {
  const setup = () =>
    render(<ActionNameCell {...mockProps} />, {
      wrapper: Table,
    });

  it('should match with snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
