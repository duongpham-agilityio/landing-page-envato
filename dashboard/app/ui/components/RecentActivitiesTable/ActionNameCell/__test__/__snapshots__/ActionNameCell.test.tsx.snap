// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionIdCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-hn5zwp"
    >
      <div
        class="css-1h2sv0j"
      >
        <p
          class="chakra-text css-1st2om9"
        >
          Login
        </p>
      </div>
    </td>
  </table>
</div>
`;

exports[`ActionNameCell component should match with snapshot 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <td
      class="css-hn5zwp"
    >
      <div
        class="css-1h2sv0j"
      >
        <p
          class="chakra-text css-1st2om9"
        >
          Login
        </p>
      </div>
    </td>
  </table>
</div>
`;
